doc/examples/affine_registration_3d.py

```diff
@@ -0,0 +1,236 @@
+solution as possible. For example, lets start with our (previously computed) rough
+transformation aligning the centers of mass of our images, and then refine it in three
+stages. First look for an optimal translation. The dictionary regtransforms contains all
+available transforms, we obtain one of them by providing its name and the dimension
+(either 2 or 3) of the image we are working with (since we are aligning volumes, the
+dimension is 3)
+"""
+
+transform = regtransforms[('TRANSLATION', 3)]
+params0 = None
+starting_affine = c_of_mass
+trans = affreg.optimize(static, moving, transform, params0,
+                        static_grid2space, moving_grid2space,
+                        starting_affine=starting_affine)
+
+"""
+If we look at the result, we can see that this translation is much better than simply
+aligning the centers of mass
+"""
+
+warped = transform_image(static, static_grid2space, moving, moving_grid2space, trans)
+regtools.overlay_slices(static, warped, None, 0, "Static", "Warped", "warped_trans_0.png")
+regtools.overlay_slices(static, warped, None, 1, "Static", "Warped", "warped_trans_1.png")
+regtools.overlay_slices(static, warped, None, 2, "Static", "Warped", "warped_trans_2.png")
+
+"""
+.. figure:: warped_trans_0.png
+   :align: center
+.. figure:: warped_trans_1.png
+   :align: center
+.. figure:: warped_trans_2.png
+   :align: center
+
+   **Registration result by just translating the moving image, using Mutual Information**.
+"""
+
+"""
+Now lets refine with a rigid transform (this may even modify our previously found
+optimal translation)
+"""
+
+transform = regtransforms[('RIGID', 3)]
+params0 = None
+starting_affine = trans
+rigid = affreg.optimize(static, moving, transform, params0,
+                        static_grid2space, moving_grid2space,
+                        starting_affine=starting_affine)
+
+"""
+This produces a slight rotation, and the images are now better aligned
+"""
+
+warped = transform_image(static, static_grid2space, moving, moving_grid2space, rigid)
+regtools.overlay_slices(static, warped, None, 0, "Static", "Warped", "warped_rigid_0.png")
+regtools.overlay_slices(static, warped, None, 1, "Static", "Warped", "warped_rigid_1.png")
+regtools.overlay_slices(static, warped, None, 2, "Static", "Warped", "warped_rigid_2.png")
+
+"""
+.. figure:: warped_rigid_0.png
+   :align: center
+.. figure:: warped_rigid_1.png
+   :align: center
+.. figure:: warped_rigid_2.png
+   :align: center
+
+   **Registration result with a rigid transform, using Mutual Information**.
+"""
+
+"""
+Finally, lets refine with a full affine transform (translation, rotation, scale and
+shear), it is safer to fit more degrees of freedom now, since we must be very close
+to the optimal transform
+"""
+
+transform = regtransforms[('AFFINE', 3)]
+params0 = None
+starting_affine = rigid
+affine = affreg.optimize(static, moving, transform, params0,
```

Comment:
Alert! We have an important difference here with the previous APIs used in your SymmetricDiffeomorphicRegistration and my StreamlineLinearRegistration. The "optimize" method should always return a Map object and that map object contains the affine (transform) and has a transform method which can transform the moving image. Here you have changed the API to only return the affine and then have independent functions. 

I think it is nice to have independent functions but it will be much nicer if you also support the initial API design.